src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -101,6 +101,20 @@ private static void WriteClientFields(CodeWriter writer, LowLevelClient client)
                 .WriteXmlDocumentationSummary($"The HTTP pipeline for sending and receiving REST requests and responses.")
                 .Line($"public virtual {typeof(HttpPipeline)} Pipeline => {client.Fields.PipelineField.Name};");
 
+            var endpointParameter = client.Parameters.FirstOrDefault(parameter => parameter.Name == "endpoint");
```

Comment:
I'm still a bit concerned that we are filtering endpoint based on the name because in some services I see we don't use endpoint name. For example: [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs#L33) in blob storage it's called url. 

@annelo-msft do we have any guideline around naming endpoint?

Reply:
Isn't BlobRestClient HLC? I didn't look into that. What is our scope? 

In LLC, the uri field is hardcoded as name pipeline at [here](https://github.com/Azure/autorest.csharp/blob/d6bd09d44c4796c0b0be12bcd1f97f526c6311ac/src/AutoRest.CSharp/LowLevel/Output/ClientFields.cs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L71).

Reply:
Yes, BlobStorage is currently HLC but our code generation should work for all the SDKs.

Could you please point me where we are hardcoding uri field?

Reply:
Yes, BlobStorage is currently HLC but our LLC code generation should work for all the SDKs.

Could you please point me where we are hardcoding uri field? If we say that the name should always be endpoint, then I fine with this :)

Reply:
> @annelo-msft do we have any guideline around naming endpoint?

I'm not aware of one - if there isn't anything in the [Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md), could you reach out to Laurent or Jeffrey?  Is there another aspect of swagger syntax we could use to identify the endpoint?

Reply:
In swagger perspective it's all string with format metadata. Endpoint parameter will be same as any other string property.

After talking to Laurent, maybe we will a linter rule in future like: "If a client level path parameter is used in a parametrized host, then it's an endpoint and it must be format as url". If that's the case then we can filter endpoint parameter based on the name and url type.

Reply:
In swagger perspective it's all string with format metadata. Endpoint parameter will be same as any other string parameter.

After talking to Laurent, maybe we will a linter rule in future like: "If a client level path parameter is used in a parametrized host, then it's an endpoint and it must be format as url". If that's the case then we can filter endpoint parameter based on the name and url type.

Reply:
> Yes, BlobStorage is currently HLC but our LLC code generation should work for all the SDKs.
> 
> Could you please point me where we are hardcoding uri field? If we say that the name should always be endpoint in SDKs, then I fine with this :)

Need to update the link to uri field [here](https://github.com/Azure/autorest.csharp/blob/90322dced7d08016294dafe9bff59a65d92c306e/src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L559).

Reply:
I'm just curious, in this code that you linked we are also hardcoding the type as Uri, then why we have endpoint generated as string [here](https://github.com/Azure/autorest.csharp/blob/45dc0b8d87c3051f6068678a198e9926ac79394c/samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs#L44) for LLC?